libs/api/domains/consultation-portal/src/lib/cases/cases.resolver.ts

```diff
@@ -0,0 +1,69 @@
+import { UseGuards } from '@nestjs/common'
+import { Args, Mutation, Query, Resolver } from '@nestjs/graphql'
+import type { User } from '@island.is/auth-nest-tools'
+import { ApiScope } from '@island.is/auth/scopes'
+import {
+  CurrentUser,
+  IdsUserGuard,
+  Scopes,
+  ScopesGuard,
+} from '@island.is/auth-nest-tools'
+import { Audit } from '@island.is/nest/audit'
+import { CaseResultService } from './cases.service'
+import { CaseResult } from '../models/caseResult.model'
+import { CaseItemResult } from '../models/caseItemResult.model'
+import { AdviceResult } from '../models/adviceResult.model'
+
+@Resolver()
+export class CaseResultResolver {
+  constructor(private caseResultService: CaseResultService) {}
+
+  @Query(() => [CaseItemResult], { name: 'consultationPortalAllCases' })
+  async getAllCases(): Promise<CaseItemResult[]> {
+    return await this.caseResultService.getAllCases()
+  }
+
+  // @Query(() => [CaseItemResult])
+  // async getCases(): Promise<CaseItemResult[]> {
+  //   return await this.caseResultService.getCase()
+  // }
+
+  @Query(() => CaseResult, { name: 'consultationPortalCaseById' })
+  async getCase(@Args('caseId') caseId: number): Promise<CaseResult> {
+    return await this.caseResultService.getCase(caseId)
+  }
+
+  @Query(() => [AdviceResult], { name: 'consultationPortalAdviceByCaseId' })
+  async getAdvices(@Args('caseId') caseId: number): Promise<string[]> {
+    const advices = await this.caseResultService.getAdvices(caseId)
+    return advices.map((advice) => advice.content as string)
+  }
+
+  @Mutation(() => CaseResult, { name: 'consultationPortalPostAdvice' })
+  async postAdvice(
+    @Args('caseId') caseId: number,
+    @Args('content') content: string,
+    @Args('files', { type: () => [String] }) files: Blob[],
+  ): Promise<void> {
```

Comment:
This resolver should probably have user authentication, get the `CurrentUser()` and forward the authentication to the backend API. With some scope check either here or in the backend.

Check out how we use @CurrentUser() in other API domains which forward it to clients.

Reply:
Will add authentication when it's ready in the backend